runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,452 @@
+      return
+    elseif not line:find("^%s*$") then
+      -- Not a blank line, not a comment, and not a block start,
+      -- so doesn't look like valid cl code.
+      break
+    end
+  end
+  vim.bo[bufnr].filetype = "dtd"
+end
+
+function M.euphoria(bufnr)
+  if vim.g.filetype_euphoria then
+    vim.bo[bufnr].filetype = vim.g.filetype_euphoria
+  else
+    vim.bo[bufnr].filetype = "euphoria3"
+  end
+end
+
+function M.ex(bufnr)
+  if vim.g.filetype_euphoria then
+    vim.bo[bufnr].filetype = vim.g.filetype_euphoria
+  else
+    for _, line in ipairs(get_lines(bufnr, 1, 100)) do
+    -- TODO: in the Vim regex, \> is used to match the end of the word, can this be omitted?
+      if line:find("^%-%-") or line:find("^ifdef") or line:find("^include")  then
+        vim.bo[bufnr].filetype = "euphoria3"
+        return
+      end
+    end
+    vim.bo[bufnr].filetype = "elixir"
+  end
+end
+
+-- This function checks the first 15 lines for appearance of 'FoamFile'
+-- and then 'object' in a following line.
+-- In that case, it's probably an OpenFOAM file
+function M.foam(bufnr)
+  local foam_file = false
+  for _, line in ipairs(get_lines(bufnr, 1, 15)) do
+    if line:find("^FoamFile") then
+      foam_file = true
+    elseif foam_file and line:find("^%s*object") then
+      vim.bo[bufnr].filetype = "foam"
+      return
+    end
+  end
+end
+
+function M.frm(bufnr)
+  if vim.g.filetype_frm then
+    vim.bo[bufnr].filetype = vim.g.filetype_frm
+  else
+    -- Always ignore case
+    local lines = get_lines(bufnr, 1, 5, { concat = true }):lower()
+    if lines:find("vb_name") or lines:find("begin vb%.form") or lines:find("begin vb%.mdiform") or lines:find("begin vb%.usercontrol") then
+        vim.bo[bufnr].filetype = "vb"
+      else
+        vim.bo[bufnr].filetype = "form"
+      end
+  end
+end
+
+function M.fs(path, bufnr)
+
+end
+
+function M.header(bufnr)
+  for _, line in ipairs(get_lines(bufnr, 1, 200)) do
+    if line:find("^@interface") or line:find("^@end") or line:find("^@class") then
+      if vim.g.c_syntax_for_h then
+        vim.bo[bufnr].filetype = "objc"
+      else
+        vim.bo[bufnr].filetype = "objcpp"
+      end
+      return
+    end
+  end
+  if vim.g.c_syntax_for_h then
+      vim.bo[bufnr].filetype = "c"
+  elseif vim.g.ch_syntax_for_h then
+      vim.bo[bufnr].filetype = "ch"
+  else
+      vim.bo[bufnr].filetype = "cpp"
+  end
+end
+
+function M.idl(bufnr)
+  for _, line in ipairs(get_lines(bufnr, 1, 50)) do
+    -- Always ignore case
+    line = line:lower()
+    if line:find('^%s*import%s+"unknwn"%.idl') or line:find('^%s*import%s+"objidl"%.idl') then
+      vim.bo[bufnr].filetype = "msidl"
+      return
+    end
+  end
+  vim.bo[bufnr].filetype = "idl"
+end
+
+function M.inc(path, bufnr)
+
+end
+
+function M.inp(bufnr)
+  if get_lines(bufnr, 1):find("^%*") then
+    vim.bo[bufnr].filetype = "abaqus"
+  else
+    for _, line in ipairs(get_lines(bufnr, 1, 500)) do
+      if line:find("^header surface data") then
```

Comment:
`dist#ft#Check_inp()` uses `=~?`

```suggestion
      if line:lower():find("^header surface data") then
```